File: Object-Oriented/Concepts/OOP.cpp
#include<iostream>

using namespace std;

class Arithmetic
{
	public: 
	
	//characteristics
	int iNo1;
	int iNo2;
	
	//behaviours
	
	Arithmetic()	//default constructor
	{
		cout<<"Inside default constructor.\n";
		iNo1 = 0;
		iNo2 = 0;
	}
	
	Arithmetic(int A, int B)	//parameterized constructor
	{
		cout<<"Inside parameterized constructor.\n";
		iNo1 = A;
		iNo2 = B;
	}

	~Arithmetic()	//destructor	
	{
		cout<<"Inside destructor.\n";
	}
	
	int Addition()
	{
		int iAns = 0;
		iAns = iNo1 + iNo2;
		return iAns;
	}
	
	int Subtraction()
	{
		int iAns = 0;
		iAns = iNo1 - iNo2;
		return iAns;
	}
};

int main(void)
{
	int iValue1 = 0, iValue2 = 0, iRet = 0;
	
	cout<<"Enter fisrt number : \n";
	cin>>iValue1;
	
	cout<<"Enter secund number : \n";
	cin>>iValue2;
		
	Arithmetic obj1;
	Arithmetic obj2(iValue1, iValue2); //paav bhaji with extra butter ani taak
	
	iRet = obj2.Addition();
	cout<<"Addition : "<<iRet<<"\n";
	
	iRet = obj2.Subtraction();
	cout<<"Subtraction : "<<iRet<<"\n";

    return 0;
}

File: MEAN_Stack/00_TypeScript/OOP.js
var Artihmetic = /** @class */ (function () {
    // constructor
    function Artihmetic(a, b) {
        this.No1 = a;
        this.No2 = b;
        // console.log("INSISDE constructor : + >" + (this.No1 + this.No2));
        console.log(this.No1 + this.No2);
    }
    // behaviours	
    Artihmetic.prototype.Addition = function () {
        var Ans = 0; // local variable
        Ans = this.No1 + this.No2;
        return Ans;
    };
    Artihmetic.prototype.Subtraction = function () {
        var Ans = 0; // local variable
        Ans = this.No1 - this.No2;
        return Ans;
    };
    return Artihmetic;
}());
var obj1 = new Artihmetic(11, 21);
var obj2 = new Artihmetic(51, 101);
var Ret = 0;
Ret = obj1.Addition();
console.log("Addition is : " + Ret);
Ret = obj1.Subtraction();
console.log("Subtraction is : " + Ret);
